base/src/lib.rs

```diff
@@ -0,0 +1,202 @@
+#![no_std]
+
+use base_io::*;
+use gstd::{msg, prelude::*, ActorId};
+use types::primitives::*;
+
+#[derive(Debug, Default)]
+pub struct Base {
+    pub issuer: ActorId,
+    pub base_type: String,
+    pub symbol: String,
+    pub parts: BTreeMap<PartId, Part>,
+}
+
+static mut BASE: Option<Base> = None;
+
+impl Base {
+    fn add_parts(&mut self, parts: BTreeMap<PartId, Part>) {
+        assert!(msg::source() == self.issuer, "wrong issuer");
+        for (part_id, part) in parts.clone() {
+            assert!(
+                self.parts.insert(part_id, part).is_none(),
+                "The part with that `PartId` already exists"
```

Comment:
I think
```suggestion
                "The part with the {part_id} `PartId` already exists"
```
or
```suggestion
                "The part with the one of given `PartId`'s already exists"
```
is better.